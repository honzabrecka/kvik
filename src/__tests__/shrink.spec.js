const {
  toRoseTrees,
  roundTowardZero,
} = require('../shrink')


describe('roundTowardZero', () => {

  it('rounds to the smallest integer greater than or equal to a given x where x < 0', () => {
    expect(roundTowardZero(-0.4)).toBe(0)
    expect(roundTowardZero(-2.1)).toBe(-2)
  })

  it('rounds to the largest integer less than or equal to a given x where x >= 0', () => {
    expect(roundTowardZero(0.4)).toBe(0)
    expect(roundTowardZero(0.6)).toBe(0)
    expect(roundTowardZero(1.7)).toBe(1)
  })

})

describe('toRoseTrees', () => {

  const [a, b] = toRoseTrees([1, 2], (x) =>Â x)

  it('toRoseTrees a', () => {
    const [value, children] = a
    expect(value).toBe(1)
    expect(children()).toBe(1)
  })

  it('toRoseTrees b', () => {
    const [value, children] = b
    expect(value).toBe(2)
    expect(children()).toBe(2)
  })

})
