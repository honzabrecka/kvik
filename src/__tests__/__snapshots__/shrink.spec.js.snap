// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`int 1`] = `
Array [
  Object {
    "children": Array [],
    "value": 0,
  },
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "value": 0,
      },
      Object {
        "children": Array [
          Object {
            "children": Array [],
            "value": 0,
          },
          Object {
            "children": Array [
              Object {
                "children": Array [],
                "value": 0,
              },
            ],
            "value": 1,
          },
        ],
        "value": 2,
      },
    ],
    "value": 3,
  },
  Object {
    "children": Array [
      Object {
        "children": Array [],
        "value": 0,
      },
      Object {
        "children": Array [
          Object {
            "children": Array [],
            "value": 0,
          },
          Object {
            "children": Array [
              Object {
                "children": Array [],
                "value": 0,
              },
            ],
            "value": 1,
          },
        ],
        "value": 2,
      },
      Object {
        "children": Array [
          Object {
            "children": Array [],
            "value": 0,
          },
          Object {
            "children": Array [
              Object {
                "children": Array [],
                "value": 0,
              },
              Object {
                "children": Array [
                  Object {
                    "children": Array [],
                    "value": 0,
                  },
                ],
                "value": 1,
              },
            ],
            "value": 2,
          },
        ],
        "value": 3,
      },
    ],
    "value": 4,
  },
]
`;
